packages/treasure/src/Treasure.test.ts

```diff
@@ -2,19 +2,66 @@ import { apply } from '@rabbitholegg/questdk/filter'
 import { describe, expect, test } from 'vitest'
 import { getAddress } from 'viem'
 import { Chains } from './utils'
-import { swap, getSupportedTokenAddresses } from './Treasure'
-import { passingTestCases, failingTestCases } from './test-transactions'
-import { V2_ROUTER_ABI } from './abi'
-import { MAGIC, ANIMA, V2_ROUTER } from './constants'
+import { mint, swap, getSupportedTokenAddresses } from './Treasure'
+import {
+  passingSwapTestCases,
+  failingSwapTestCases,
+  MINT_TREASURE_TAG,
+  passingMintTestCases,
+  failingMintTestCases
+} from './test-transactions'
+import {MINT_TREASURE_TAG_ABI, V2_ROUTER_ABI} from './abi'
+import {MAGIC, ANIMA, V2_ROUTER} from './constants'
+
+describe('Given the treasure plugin', () => {
+  describe('when handling the mint action', () => {
+    describe('should return a valid action filter', () => {
+      const { params } = MINT_TREASURE_TAG
+      test('when minting a Treasure Tag through the proxy contract', async () => {
+        const filter = await mint(params)
+        expect(filter).to.deep.equal({
+          chainId: Chains.ARBITRUM_ONE,
+          to: params.contractAddress,
+          from: params.recipient,
+          input: {
+            $abi: MINT_TREASURE_TAG_ABI,
+            _registerArgs: {
+              owner: params.recipient,
```

Comment:
is this `owner` param redundant given that it will always be the same as the `from` param?